### PR TITLE
Sprint b01   create user

### PR DESCRIPTION
**Pull request by DDanielVieira**

### CODE REVIEW

File: src/main/java/com/criticalsoftware/Contact.java [hunk 1]
```diff
@@ -1,51 +1,53 @@
...
-
 import io.quarkus.mongodb.panache.PanacheMongoEntity;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
 
 public class Contact extends PanacheMongoEntity {
 
+    @Email(message = "Email should be valid")
+    @NotBlank(message = "Email may not be blank")
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Contact.java:13]: Cannot be null, also.
@NotNull
Use the same on address, please.

File: src/main/java/com/criticalsoftware/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import io.quarkus.mongodb.panache.PanacheMongoRepository;
+import io.quarkus.panache.common.Sort;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.List;
+
+@ApplicationScoped
+public class UserRepository implements PanacheMongoRepository<User> {
+
+    // Este método procura um usuário pelo campo 'name'
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/UserRepository.java]: In code or comments, only english languaged is allowed.

File: src/main/java/com/criticalsoftware/UserRequest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import java.time.LocalDate;
+
+//User Data Request
+public class UserRequest {
+    @NotBlank(message = "Name may not be blank")
+    private String name;
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/UserRequest.java:11]: Not null missing.

File: src/main/java/com/criticalsoftware/UserRequest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    @NotNull(message ="Date Birth is mandatory")
+    private LocalDate dateBirth;
+
+    @Valid
+    @NotNull(message = "Contact is mandatory")
+    private Contact contact;
+
+    public UserRequest() {
+        // Não precisa necessariamente de implementação aqui
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/UserRequest.java]: True, but in english, please.

**Review by jpmoreira-training — CHANGES_REQUESTED** [review]

File: src/main/java/com/criticalsoftware/Contact.java (hunk 1 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/Contact.java:13]: The @NotBlank ensures that a String is not null, is not empty, and has at least one visible character.

File: src/main/java/com/criticalsoftware/UserRepository.java (hunk 2 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/UserRepository.java]: Translation done

File: src/main/java/com/criticalsoftware/UserRequest.java (hunk 4 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/UserRequest.java]: Translation done

File: src/main/java/com/criticalsoftware/UserRequest.java (hunk 3 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/UserRequest.java:11]: The @notblank ensures that a String is not null, is not empty, and has at least one visible character.

**Pull request merged**